### PR TITLE
feat: Public Delegate List

### PR DESCRIPTION
**Pull request by katamarinaki**
### Changelog

- Added new component to display public delegates list

- Refactored DelegatesList to improve code readability and reduce complexity.

### CODE REVIEW

File: modules/delegation/providers/DelegationFormContext.tsx [hunk 1]
```diff
@@ -116,17 +118,36 @@ const useDelegationFormActions = (
...
   })
 
+  useEffect(() => {
+    const currentValue = formObject.getValues('delegateAddress')
+    if (
+      presetDelegateAddress &&
+      isValidAddress(presetDelegateAddress) &&
+      currentValue?.toLowerCase() !== presetDelegateAddress.toLowerCase()
+    ) {
+      console.log('setting default address', presetDelegateAddress)
```
Comment (rkolpakov) [inline modules/delegation/providers/DelegationFormContext.tsx:145]: Is that intended?

**Review by rkolpakov — DISMISSED** [review]
LGTM

File: modules/delegation/providers/DelegationFormContext.tsx (hunk 1 above)
Comment (katamarinaki) [reply modules/delegation/providers/DelegationFormContext.tsx:145]: No, of course not, thanks for pointing out

File: modules/shared/hooks/useEnsResolvers.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const { getRpcUrl } = useConfig()
+
+  const ethProvider = useMemo(() => {
+    const rpcUrl = getRpcUrl(chainId)
+    return getStaticRpcBatchProvider(chainId, rpcUrl)
+  }, [chainId, getRpcUrl])
+
+  const lookupAddress = useCallback(
+    async (address: string) => {
+      // ENS name is not supported on Holesky for our current setup
```
Comment (BATMAH69) [inline modules/shared/hooks/useEnsResolvers.ts:18]: There ensName hook witch works with holeky
https://github.com/lidofinance/dao-voting-ui/blob/develop/modules/shared/hooks/useEnsNames.ts

**Review by BATMAH69 — APPROVED** [review]

File: modules/shared/hooks/useEnsResolvers.ts (hunk 2 above)
Comment (katamarinaki) [reply modules/shared/hooks/useEnsResolvers.ts:18]: We'll merge those in the next release

File: modules/delegation/publicDelegates.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    address: '0x04827a54F2e345467beAFEfB9EF76Cb2f2c62D83',
+    lido: 'https://research.lido.fi/t/daedalus-delegate-thread/8195',
+    twitter: 'https://x.com/daedalus_angels',
+  },
+  {
+    name: 'ProRelGuild',
+    avatar: null,
+    address: 'prorelguild.eth',
+    lido: 'https://research.lido.fi/t/prorelguild-delegate-thread/8186',
+    twitter: 'https://x.com/apegenija',
```
[the remark this answers is not in the document]
Comment (katamarinaki) [reply modules/delegation/publicDelegates.ts:207]: Thanks!

**Pull request merged**